src/main/java/org/jabref/logic/importer/fileformat/MsBibImporter.java

```diff
@@ -24,24 +25,25 @@
 /**
  * Importer for the MS Office 2007 XML bibliography format
  */
+@NullMarked
 public class MsBibImporter extends Importer {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(MsBibImporter.class);
     private static final String DISABLEDTD = "http://apache.org/xml/features/disallow-doctype-decl";
     private static final String DISABLEEXTERNALDTD = "http://apache.org/xml/features/nonvalidating/load-external-dtd";
+    private static final DocumentBuilderFactory DOCUMENT_BUILDER_FACTORY = makeSafeDocBuilderFactory(DocumentBuilderFactory.newInstance());
```

Comment:
The DocumentBuilderFactory as a constant? This smells funny, but also suspicious...

Reply:
This has performance reasons. A Factory is instantiated once, but the produced things are instantiated more often (by the factory). - It takes time to setup the factory.

See https://community.sap.com/t5/technology-blogs-by-members/performance-improvements-in-nw-java-applications-with-xml-processing/ba-p/12894611 for more information.

> Solution - Make XML Factories Static